File: pertemuan9/Prak/PRAK01_2473021_JAVA/C_ChaesarChipper.java
/**
 * Pertemuan 09
 * [PRAKTIKUM]  Program Gambaran sederhana tentang enkripsi, misalnya mengganti huruf a dengan j, b dengan k dan seterusnya
 *
 * @author 2473021-Febrianus Leona Putra
 * @version 19 Oktober 2024
 */
package pertemuan9.Prak.PRAK01_2473021_JAVA;

import java.util.Scanner;

public class C_ChaesarChipper {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Masukkan Kalimat: ");
        String kalimat = sc.nextLine();

        char[] tabelSubstitusi = {
            'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S',
            'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'A', 'B', 'C',
            'D', 'E', 'F', 'G', 'H', 'I'
        };

        StringBuilder terenkripsi = new StringBuilder();

        for (char c : kalimat.toCharArray()) {
            if (Character.isLetter(c)) {
                if (Character.isUpperCase(c)) {
                    terenkripsi.append(tabelSubstitusi[c - 'A']);
                } else {
                    terenkripsi.append(Character.toLowerCase(tabelSubstitusi[c - 'a']));
                }
            } else if (c == ' ') {
                terenkripsi.append('%');
            } else {
                terenkripsi.append('$');
            }
        }

        String teksTerenkripsi = terenkripsi.toString();
        System.out.println("Hasil Enkripsi: " + teksTerenkripsi);

        sc.close();
    }
}

File: pertemuan9/Prak/PRAK01_2473021_PY/C_ChaesarChipper.py
"""
Pertemuan 09
[PRAKTIKUM]  Program Gambaran sederhana tentang enkripsi, misalnya mengganti huruf a dengan j, b dengan k dan seterusnya

@author 2473021-Febrianus Leona Putra
@version 19 Oktober 2024
"""
teks = input("Masukkan Kalimat: ")

tabelsubstitusi = {
    'A': 'J', 'B': 'K', 'C': 'L', 'D': 'M', 'E': 'N', 'F': 'O', 'G': 'P', 'H': 'Q', 'I': 'R',
    'J': 'S', 'K': 'T', 'L': 'U', 'M': 'V', 'N': 'W', 'O': 'X', 'P': 'Y', 'Q': 'Z', 'R': 'A',
    'S': 'B', 'T': 'C', 'U': 'D', 'V': 'E', 'W': 'F', 'X': 'G', 'Y': 'H', 'Z': 'I'
}


terenkripsi = []
for karakter in teks:
    if karakter.isalpha():
        if karakter.isupper():
            terenkripsi.append(tabelsubstitusi[karakter])
        else:
            terenkripsi.append(tabelsubstitusi[karakter.upper()].lower())
    elif karakter == ' ':
        terenkripsi.append('%')
    else:
        terenkripsi.append('$')

teksterenkripsi = ''.join(terenkripsi)
print("Hasil Enkripsi:", teksterenkripsi)
